ons_alpha/bulletins/models.py

```diff
@@ -58,3 +60,23 @@ class BulletinPage(BasePage):
             ObjectList(BasePage.settings_panels, heading="Settings", classname="settings"),
         ]
     )
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+
+        context["toc"] = self.toc
+
+        return context
+
+    @cached_property
+    def toc(self):
+        items = [{"url": "#summary", "text": "Summary"}]
+
+        for block in self.body:  # pylint: disable=not-an-iterable
+            if hasattr(block.block, "to_table_of_contents_items"):
+                items += block.block.to_table_of_contents_items(block.value)
+
+        if self.contact_details_id:
+            items += [{"url": "#contact-details", "text": "Contact details"}]
```

Comment:
@mr-nick17 this is the approach to add various elements to the TOC

the `table_of_contents` template tag (`{{ table_of_contents(page, "body", {"title": "Table of contents"}) }}`) currently only picks the up from a StreamField, rather than anything else